nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -164,6 +159,11 @@ protected DocumentModel createEntryWithoutReferences(Map<String, Object> fieldMa
             }
         }
 
+        // Check if the entry already exists
+        if (hasEntry(String.valueOf(fieldMap.get(idFieldName)))) {
```

Comment:
Why not re-using id like you did in `SQLSession` ?

Reply:
Changed to reuse id